File: hejoo345/week_22/pgs_단속카메라.js
function solution(routes) {
    let answer = 0;
    routes.sort((a, b) => a[1] - b[1]);
    let camera = -30001;
    routes.forEach(route => {
        if (route[0] > camera) {
            answer++;
            camera = route[1];
        }
    });
    return answer;
}

File: OneK-2/week_22/PGS_단속카메라.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        Arrays.sort(routes, Comparator.comparingInt(o -> o[1]));
        int min = Integer.MIN_VALUE;

        for (int[] r : routes) {
            if(min < r[0]){
                min = r[1];
                answer++;
            }
        }

        return answer;
    }
}